File: java/javasearch/src/test/java/javasearch/FileTypesTest.java
package javasearch;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import java.io.File;
import org.junit.Test;

public class FileTypesTest {
    private final FileTypes fileTypes;

    public FileTypesTest() {
        fileTypes = new FileTypes();
    }

    @Test
    public final void archiveFileTest() {
        File file = new File("test.zip");
        assertEquals("zip", FileUtil.getExtension(file));
        assertTrue(fileTypes.isArchiveFile(file));
        assertFalse(fileTypes.isBinaryFile(file));
        assertFalse(fileTypes.isCodeFile(file));
        assertTrue(fileTypes.isSearchableFile(file));
        assertFalse(fileTypes.isTextFile(file));
        assertFalse(fileTypes.isXmlFile(file));
        assertEquals(FileType.ARCHIVE, fileTypes.getFileType(file));
    }

    @Test
    public final void binaryFileTest() {
        File file = new File("test.exe");
        assertEquals("exe", FileUtil.getExtension(file));
        assertFalse(fileTypes.isArchiveFile(file));
        assertTrue(fileTypes.isBinaryFile(file));
        assertFalse(fileTypes.isCodeFile(file));
        assertTrue(fileTypes.isSearchableFile(file));
        assertFalse(fileTypes.isTextFile(file));
        assertFalse(fileTypes.isXmlFile(file));
        assertEquals(FileType.BINARY, fileTypes.getFileType(file));
    }

    @Test
    public final void codeFileTest() {
        File file = new File("Test.java");
        assertEquals("java", FileUtil.getExtension(file));
        assertFalse(fileTypes.isArchiveFile(file));
        assertFalse(fileTypes.isBinaryFile(file));
        assertTrue(fileTypes.isCodeFile(file));
        assertTrue(fileTypes.isSearchableFile(file));
        assertTrue(fileTypes.isTextFile(file));
        assertFalse(fileTypes.isXmlFile(file));
        assertEquals(FileType.CODE, fileTypes.getFileType(file));
    }

    @Test
    public final void textFileTest() {
        File file = new File("test.txt");
        assertEquals("txt", FileUtil.getExtension(file));
        assertFalse(fileTypes.isArchiveFile(file));
        assertFalse(fileTypes.isBinaryFile(file));
        assertFalse(fileTypes.isCodeFile(file));
        assertTrue(fileTypes.isSearchableFile(file));
        assertTrue(fileTypes.isTextFile(file));
        assertFalse(fileTypes.isXmlFile(file));
        assertEquals(FileType.TEXT, fileTypes.getFileType(file));
    }

    @Test
    public final void xmlFileTest() {
        File file = new File("markup.xml");
        assertEquals("xml", FileUtil.getExtension(file));
        assertFalse(fileTypes.isArchiveFile(file));
        assertFalse(fileTypes.isBinaryFile(file));
        assertFalse(fileTypes.isCodeFile(file));
        assertTrue(fileTypes.isSearchableFile(file));
        assertTrue(fileTypes.isTextFile(file));
        assertTrue(fileTypes.isXmlFile(file));
        assertEquals(FileType.XML, fileTypes.getFileType(file));
    }

    @Test
    public final void unknownFileTest() {
        File file = new File("unknown.ZZZ");
        assertEquals("zzz", FileUtil.getExtension(file));
        assertFalse(fileTypes.isArchiveFile(file));
        assertFalse(fileTypes.isBinaryFile(file));
        assertFalse(fileTypes.isSearchableFile(file));
        assertFalse(fileTypes.isTextFile(file));
        assertFalse(fileTypes.isXmlFile(file));
        assertTrue(fileTypes.isUnknownFile(file));
        assertEquals(FileType.UNKNOWN, fileTypes.getFileType(file));
    }
}

File: kotlin/ktsearch/src/test/kotlin/ktsearch/FileTypesTest.kt
package ktsearch

import org.junit.Test
import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import java.io.File

/**
 * @author cary on 7/30/16.
 */
class FileTypesTest {
    private val fileTypes = FileTypes()

    @Test
    fun archiveFileTest() {
        val file = File("test.zip")
        assertEquals("zip", file.extension)
        assertTrue(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isSearchableFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.ARCHIVE, fileTypes.getFileType(file))
    }

    @Test
    fun binaryFileTest() {
        val file = File("test.exe")
        assertEquals("exe", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertTrue(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isSearchableFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.BINARY, fileTypes.getFileType(file))
    }

    @Test
    fun javaFileTest() {
        val file = File("Test.java")
        assertEquals("java", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertTrue(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isSearchableFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.TEXT, fileTypes.getFileType(file))
    }

    @Test
    fun textFileTest() {
        val file = File("test.txt")
        assertEquals("txt", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isSearchableFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.TEXT, fileTypes.getFileType(file))
    }

    @Test
    fun textFileTestUppercase() {
        val file = File("TEXT.TXT")
        assertEquals("TXT", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertTrue(fileTypes.isSearchableFile(file))
        assertTrue(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertEquals(FileType.TEXT, fileTypes.getFileType(file))
    }

    @Test
    fun unknownFileTest() {
        val file = File("unknown.ZZZ")
        assertEquals("ZZZ", file.extension)
        assertFalse(fileTypes.isArchiveFile(file))
        assertFalse(fileTypes.isBinaryFile(file))
        assertFalse(fileTypes.isCodeFile(file))
        assertFalse(fileTypes.isSearchableFile(file))
        assertFalse(fileTypes.isTextFile(file))
        assertFalse(fileTypes.isXmlFile(file))
        assertTrue(fileTypes.isUnknownFile(file))
        assertEquals(FileType.UNKNOWN, fileTypes.getFileType(file))
    }
}
